Modules/Tests/YosemiteTests/Tools/POS/POSCatalogSyncCoordinatorTests.swift

```diff
@@ -860,4 +860,78 @@ extension POSCatalogSyncCoordinatorTests {
         // Then - sync should proceed (exactly at 30-day boundary is still eligible)
         #expect(mockIncrementalSyncService.startIncrementalSyncCallCount == 1)
     }
+
+    // MARK: - isSyncStale Tests
+
+    @Test func test_isSyncStale_returns_true_when_no_full_sync_performed() async throws {
+        // Given - no full sync date set
+
+        // When
+        let isStale = await sut.isSyncStale(for: sampleSiteID, maxDays: 7)
+
+        // Then
+        #expect(isStale == true)
+    }
+
+    @Test func test_isSyncStale_returns_false_when_full_sync_is_recent() async throws {
+        // Given - last full sync was 3 days ago
+        let threeDaysAgo = Calendar.current.date(byAdding: .day, value: -3, to: Date())!
+        try createSiteInDatabase(siteID: sampleSiteID, lastFullSyncDate: threeDaysAgo)
+
+        // When
+        let isStale = await sut.isSyncStale(for: sampleSiteID, maxDays: 7)
+
+        // Then
+        #expect(isStale == false)
+    }
+
+    @Test func test_isSyncStale_returns_true_when_full_sync_is_old() async throws {
+        // Given - last full sync was 10 days ago
+        let tenDaysAgo = Calendar.current.date(byAdding: .day, value: -10, to: Date())!
+        try createSiteInDatabase(siteID: sampleSiteID, lastFullSyncDate: tenDaysAgo)
+
+        // When
+        let isStale = await sut.isSyncStale(for: sampleSiteID, maxDays: 7)
+
+        // Then
+        #expect(isStale == true)
+    }
+
+    @Test func test_isSyncStale_ignores_incremental_sync_date() async throws {
+        // Given - incremental sync was recent, but full sync was old
+        let yesterday = Calendar.current.date(byAdding: .day, value: -1, to: Date())!
+        let tenDaysAgo = Calendar.current.date(byAdding: .day, value: -10, to: Date())!
+        try createSiteInDatabase(siteID: sampleSiteID, lastFullSyncDate: tenDaysAgo, lastIncrementalSyncDate: yesterday)
+
+        // When
+        let isStale = await sut.isSyncStale(for: sampleSiteID, maxDays: 7)
+
+        // Then - should only check full sync date
+        #expect(isStale == true)
+    }
+
+    @Test func test_isSyncStale_boundary_within_threshold() async throws {
+        // Given - last full sync was 6 days and 23 hours ago (just under 7 days)
+        let justUnderSevenDays = Calendar.current.date(byAdding: .day, value: -6, to: Date())!
+            .addingTimeInterval(-23 * 60 * 60) // minus 23 hours
+        try createSiteInDatabase(siteID: sampleSiteID, lastFullSyncDate: justUnderSevenDays)
+
+        // When
+        let isStale = await sut.isSyncStale(for: sampleSiteID, maxDays: 7)
+
+        // Then - just under threshold should not be stale
+        #expect(isStale == false)
+    }
+
+    @Test func test_isSyncStale_boundary_past_threshold() async throws {
```

Comment:
nit: this test case feels similar to `test_isSyncStale_returns_true_when_full_sync_is_old`, maybe the last full sync could be like 1 second past the threshold or exactly at the threshold?